### PR TITLE
Make json converters strongly typed.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Also, add a json converter for TextChange, it will be needed for AddUsing-OOP.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide  @heejaechang

**Comment by CyrusNajmabadi** [comment]
nuget cache corruption: https://ci.dot.net/job/dotnet_roslyn/job/master/job/windows_release_unit64_prtest/6980/

**Comment by CyrusNajmabadi** [comment]
retest windows_release_unit64_prtest  please

**Comment by CyrusNajmabadi** [comment]
Telling @MattGertz This is part of our OOP work.  It makes core serialization code more typesafe so that there aren't accidently conversions between incompatible types.

**Comment by CyrusNajmabadi** [comment]
CompilerServerUnitTests.ExtraMSCorLibCS failed

**Comment by CyrusNajmabadi** [comment]
retest windows_release_unit64_prtest please

File: src/Workspaces/Remote/ServiceHub/Shared/RoslynJsonConverter.cs [hunk 1]
```diff
@@ -55,34 +55,43 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
...
-            protected static T ReadProperty<T>(JsonSerializer serializer, JsonReader reader)
+            public sealed override bool CanConvert(Type objectType) => typeof(T) == objectType;
+
+            public sealed override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+                => ReadValue(reader, objectType, existingValue, serializer);
+
+            public sealed override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+                => WriteValue(writer, (T)value, serializer);
+
+            protected abstract T ReadValue(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer);
```
Comment (Pilchie) [inline src/Workspaces/Remote/ServiceHub/Shared/RoslynJsonConverter.cs]: Does anyone use `objectType` or `existingValue`?  Consider removing to simplify if we don't need it?

**Comment by heejaechang** [comment]
nice 👍

**Pull request merged**